Watcher isn't Sync with --no-default-features

>> Note: we may want to document and test for this, so it isn't reverted later on with a trait under the impression of usability improvements
>
>I did add a simple test for this in src/lib.rs, where I assert that the `Watcher` trait is object safe. So we can't accidentally remove this feature.

_Originally posted by @erickt in https://github.com/notify-rs/notify/issues/336#issuecomment-858929800_

That only works when not using `std::sync::mpsc::Sender`:

```
$ cargo test -p notify --no-default-features
   Compiling notify v6.0.1 (notify/notify)
error[E0277]: `std::sync::mpsc::Sender<EventLoopMsg>` cannot be shared between threads safely
   --> notify/src/inotify.rs:570:13
    |
570 |     check::<INotifyWatcher>();
    |             ^^^^^^^^^^^^^^ `std::sync::mpsc::Sender<EventLoopMsg>` cannot be shared between threads safely
    |
    = help: within `INotifyWatcher`, the trait `Sync` is not implemented for `std::sync::mpsc::Sender<EventLoopMsg>`
note: required because it appears within the type `INotifyWatcher`
   --> notify/src/inotify.rs:45:12
    |
45  | pub struct INotifyWatcher {
    |            ^^^^^^^^^^^^^^
note: required by a bound in `inotify::inotify_watcher_is_send_and_sync::check`
   --> notify/src/inotify.rs:569:24
    |
569 |     fn check<T: Send + Sync>() {}
    |                        ^^^^ required by this bound in `inotify::inotify_watcher_is_send_and_sync::check`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `notify` due to previous error
```

--- a/notify-debouncer-full/src/lib.rs
+++ b/notify-debouncer-full/src/lib.rs
@@ -16,7 +16,7 @@
 //! ```
 //!
 //! In case you want to select specific features of notify,
-//! specify notify as dependency explicitely in your dependencies.
+//! specify notify as dependency explicitly in your dependencies.
 //! Otherwise you can just use the re-export of notify from debouncer-full.
 //!
 //! ```toml

--- a/notify-debouncer-mini/src/lib.rs
+++ b/notify-debouncer-mini/src/lib.rs
@@ -7,7 +7,7 @@
 //! notify-debouncer-mini = "0.4.0"
 //! ```
 //! In case you want to select specific features of notify,
-//! specify notify as dependency explicitely in your dependencies.
+//! specify notify as dependency explicitly in your dependencies.
 //! Otherwise you can just use the re-export of notify from debouncer-mini.
 //! ```toml
 //! notify-debouncer-mini = "0.4.0"

--- a/notify/src/lib.rs
+++ b/notify/src/lib.rs
@@ -21,7 +21,7 @@
 //!
 //! ### Serde
 //!
-//! Events are serialisable via [serde](https://serde.rs) if the `serde` feature is enabled:
+//! Events are serializable via [serde](https://serde.rs) if the `serde` feature is enabled:
 //!
 //! ```toml
 //! notify = { version = "6.0.1", features = ["serde"] }
@@ -29,7 +29,7 @@
 //!
 //! ### Crossbeam-Channel & Tokio
 //!
-//! By default crossbeam-channel is used internally by notify.
+//! By default crossbeam-channel is used internally by notify. Which also allows the [Watcher] to be sync.
 //! This can [cause issues](https://github.com/notify-rs/notify/issues/380) when used inside tokio.
 //!
 //! You can disable crossbeam-channel, letting notify fallback to std channels via
@@ -38,9 +38,16 @@
 //! notify = { version = "6.0.1", default-features = false, features = ["macos_kqueue"] }
 //! // Alternatively macos_fsevent instead of macos_kqueue
 //! ```
-//! Note the `macos_kqueue` requirement here, otherwise no backend is available on macos.
+//! Note the `macos_kqueue` requirement here, otherwise no native backend is available on macos.
 //!
 //! # Known Problems
+//! 
+//! ### Network filesystems
+//! 
+//! Network mounted filesystems like NFS may not emit any events for notify to listen to.
+//! This applies especially to WSL programs watching windows paths ([issue #254](https://github.com/notify-rs/notify/issues/254)).
+//! 
+//! A workaround is the [PollWatcher] backend.
 //!
 //! ### Docker with Linux on MacOS M1
 //!
@@ -65,7 +72,7 @@
 //! If you want to receive an event for a deletion of folder `b` for the path `/a/b/..`, you will have to watch its parent `/a`.
 //! See [here](https://github.com/notify-rs/notify/issues/403) for more details.
 //!
-//! ### Pseudo Filesystems like /proc,/sys
+//! ### Pseudo Filesystems like /proc, /sys
 //!
 //! Some filesystems like `/proc` and `/sys` on *nix do not emit change events or use correct file change dates.
 //! To circumvent that problem you can use the [PollWatcher] with the `compare_contents` option.
@@ -84,6 +91,11 @@
 //!
 //! Note that the [PollWatcher] is not restricted by this limitation, so it may be an alternative if your users can't increase the limit.
 //!
+//! ### Watching large directories
+//! 
+//! When watching a very large amount of files, notify may fail to receive all events.
+//! For example the linux backend is documented to not be a 100% reliable source. See also issue [#412](https://github.com/notify-rs/notify/issues/412).
+//! 
 //! # Examples
 //!
 //! For more examples visit the [examples folder](https://github.com/notify-rs/notify/tree/main/examples) in the repository.

